lib/ApacheModules.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * ownCloud
+ *
+ * @author Saugat Pachhai <saugat@jankaritech.com>
+ * @copyright Copyright (c) 2019 Saugat Pachhai saugat@jankaritech.com
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Testing;
+
+use OC\OCS\Result;
+use OCP\API;
+use OCP\IRequest;
+
+class ApacheModules {
+	/**
+	 *
+	 * @var IRequest
+	 */
+	private $request;
+
+	/**
+	 *
+	 * @param IRequest $request
+	 */
+	public function __construct(IRequest $request) {
+		$this->request = $request;
+	}
+
+	protected function isApache() {
+		return (\strpos($_SERVER['SERVER_SOFTWARE'], 'Apache') !== false);
+	}
+
+	protected function getModules() {
```

Comment:
why not returning all modules and let the client decide what to do with it

Reply:
This really simplifies the implementation (see: owncloud/core#35401). If we really need to get all the modules, we can have another api (eg: `/apache_modules`). I thought it'd be better to handle it here rather than at our test runner.